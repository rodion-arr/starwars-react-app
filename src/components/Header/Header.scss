.header {
  position: fixed;
  top: 0;
  display: flex;
  align-content: center;
  justify-content: space-between;
  height: 60px;
  width: 100%;
  background: #000000;
  border-bottom: solid 1px var(--border-color-light);

  &__logo {
    background: no-repeat center/200px url("../../images/header/sw_logo_horizontal.png");
    width: 200px;
  }
}

.burger-menu {
  box-sizing: border-box;
  width: 60px;
  padding: 10px;
  display: flex;
  flex-direction: column;
  justify-content: space-around;
  border-right: solid 1px var(--border-color-light);

  &__bar {
    border: solid 2px var(--border-color-light);
  }
}

.search {
  width: 60px;
  border-left: solid 1px var(--border-color-light);
  display: flex;
  justify-content: center;

  svg {
    fill: var(--border-color-light);
    width: 30px;
  }
}
