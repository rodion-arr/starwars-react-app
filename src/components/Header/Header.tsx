import React from "react";
import './Header.scss';
import {MainMenu} from "../MainMenu/MainMenu";
import {Link} from "react-router-dom";
import {URI} from "../../routes/uri";

export const Header = (): JSX.Element => {
  return <div className="header">
    <MainMenu />
    <Link to={URI.home} className="header__logo" />
    <div className="header__fork fork">
      <a
        href="https://github.com/rodion-arr/starwars-react-app"
        title="Fork me on GitHub"
        target="_blank"
        rel="noreferrer"
      ><i className="fork__github-icon" /></a>
    </div>
  </div>
}
