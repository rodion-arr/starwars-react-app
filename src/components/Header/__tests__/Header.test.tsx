import {render} from "@testing-library/react";
import {Header} from "../Header";
import { BrowserRouter as Router } from "react-router-dom"

describe('Header', () => {
  it('should render correctly', function () {
    const {container} = render(<Router><Header /></Router>);

    expect(container).toMatchSnapshot();
  });
});
