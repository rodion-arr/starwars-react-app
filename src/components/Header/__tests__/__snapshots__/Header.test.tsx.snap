// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render correctly 1`] = `
<div>
  <div
    class="header"
  >
    <div
      class="header__menu burger-menu"
    >
      <div
        class="burger-menu__bar"
      />
      <div
        class="burger-menu__bar"
      />
      <div
        class="burger-menu__bar"
      />
    </div>
    <a
      class="header__logo"
      href="/"
    />
    <div
      class="header__fork fork"
    >
      <a
        href="https://github.com/rodion-arr/starwars-react-app"
        rel="noreferrer"
        target="_blank"
        title="Fork me on GitHub"
      >
        <i
          class="fork__github-icon"
        />
      </a>
    </div>
  </div>
</div>
`;
